src/app/filter_utils.ts

```diff
@@ -137,10 +141,13 @@ const MILLISECONDS_PER_DAY = 1000 * 60 * 60 * 24;
 // of yesterday (midnight yesterday), but this could be interpreted 
 // differently. We're also interpreting weeks and months relative to today, 
 // but these could also be determined based on where we are in the month.
-export function buildDateFilterForNDays(now: Date, days: number): DateFilter {
-  const midnightToday = new Date(now).setHours(0, 0, 0, 0);
+// Added an optional filterType parameter to allow for the option of using the current time as the start time instead of midnight.
+export function buildDateFilterForNDays(now: Date, days: number, fromFilter: DayFilterType = DayFilterType.MIDNIGHT): DateFilter {
+
+  const fromTime = fromFilter === DayFilterType.NOW ? now.getTime() : new Date(now).setHours(0, 0, 0, 0);
```

Comment:
```suggestion
  const fromTime = fromFilter === DayFilterType.NOW ? now.getTime() : new Date(now).setHours(0, 0, 0, 0);
```

Reply:
@kilemensi Not sure what change is here as the suggestion looks similar to what is already there